src/transformState.js

```diff
@@ -62,7 +62,33 @@
  * @param {Object[]} transforms
  */
 function transformState(state, transforms) {
-  // write code here
+  let addPror;
+
+  for (const objTrans of transforms) {
+    addPror = objTrans;
+
+    if (addPror.operation === 'addProperties') {
+      Object.assign(state, addPror.properties);
+    }
+
+    if (addPror.operation === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
+
+    if (addPror.operation === 'removeProperties') {
+      for (const removeKey of addPror.properties) {
+        for (const key in state) {
+          if (removeKey === key) {
+            delete state[key];
+          }
+        }
+      }
```

Comment:
you need only one loop here, if state hasn't this key, it's do nothing

Reply:
I don't understand, that should I change. If I'm using one loop, it doesn't work correctly

Reply:
for (const removeKey of addPror.properties) {
  delete state[removeKey];
}